DashAISoftware/DashAI
# Validation of input and output columns

# Summary


Creates an endpoint to validate input and output columns with the selected task requierements and shows if they are valid on the prepare dataset step.

## Type of change



- Front end new feature.
- Back end new feature.

## Changes


- New endpoint experiments/validation/ with HTTP POST method.
- Adds unit tests for the new endpoint.
- Adds logic in Prepare Dataset Step to update the warning with the validation answer.

## How to Test


- To test the endpoint you have to run ```pytest tests/back/api/test_experiments_api.py```
- To see the Prepare Dataset Step and its new feature:
1.- Run ```python -m DashAI```
2.- Go to ```dashai/front``` and run ```yarn start```
3.- When the localhost starts on your browser go to "New Experiment" and follow the steps until you get to Prepare Dataset Step.

## Screenshots


![image](https://github.com/DashAISoftware/DashAI/assets/70553082/13f37156-c8cf-4c7d-8ea2-8eb35604f604)
![image](https://github.com/DashAISoftware/DashAI/assets/70553082/25a27f45-e1d2-4c4d-92ea-6c3e5800a21c)
